### PR TITLE
Rework codebase to reflect v2 authentication

### PR DESCRIPTION
**Pull request by markusthoemmes**
This pull-request intends to reflect the new way of authentication with namespace bound keys throughout the codebase.

### CODE REVIEW

**Comment by ddragosd** [comment]
@markusthoemmes I'm trying to understand what is the advantage of coupling `identity` (represented by the AuthKey) with the `namespaces`.  I thought the AuthKey is used to authenticate a user/developer and that whether a user is authorized to manage a namespace or not is another concern. Can you share more details to help me understand ?

**Comment by markusthoemmes** [comment]
@ddragosd in a nutshell: Using the old keys we had a hard time associating a namespace to the current usage, as in: To which namespace should the activation of the current invocation go? The namespace of the action? Doesn't work for `whisk.system` actions for example.

We decided the best solution would be to always identify as a user + a namespace, hence each key is associated with a namespace now.

For posterity: Here's the initial proposal: #1107

**Comment by ddragosd** [comment]
thanks @markusthoemmes.

Am I reading correctly that the assumption is that the developer creating the action is also the one invoking the action ? How about actions that are exposed as REST endpoints via the API Gateway which have no OW AuthKey, but have only an OAuth 2 Token from a 3rd party identity provider ? How would we connect these OAuth tokens with some OW credentials used to deploy the action so that we can invoke it ? I think we have to solve this problem if we go down this path. WDYT ? 

This is connected to another aspect that I struggle to understand in OW and that is the separation between the _management_ aspect ( the developers creating and managing actions ) vs the _execution_ aspect (the end-users of an application that invokes actions). In my mind these 2 categories : developers vs end-users have different concerns. They authenticate differently, end-users might not care about namespaces, logs, package names, actions names.

Thinking at the access to the activations and logs of an action, I imagine it's something that concerns more the team that is maintaining that action. If an application that uses an action gets an error back, even though it can access the activations and logs, but if it can't fix the action anyway, what's the utility other than debugging that action and maybe billing ?  

I feel like we're missing another entity that would represent "apps". "Apps" are part of the _execution_ concern b/c they invoke the actions. Apps would subscribe to actions to be able to invoke them. Application info is part of the OAuth token ( the `client_id` that has generated the token ). When an OAuth token is not present we can ask the application to identity itself using an API-KEY for instance.  This way we could show to multiple personas information about activations, _who_ has invoked actions, and _what_ actions. In this way we don't have to solve the problem of matching an OAuth token with an OW Auth Key anymore. We're solving a different problem instead and we can also give access to multiple people to see the analytics for an application. Based on the usage data we can also do billing or charge-backs.

**Comment by rabbah** [comment]
> Am I reading correctly that the assumption is that the developer creating the action is also the one invoking the action ?

No of course not. Any authorized subject can invoke their actions (in their namespace which may be a shared space with other subjects) or actions in another namespace if the action is in a public package.

**Comment by rabbah** [comment]
It's worth noting this change is semantic preserving for openwhisk.

File: ansible/templates/whisk.properties.j2 [hunk 1]
```diff
@@ -14,6 +14,8 @@ whisk.ssl.cert={{ openwhisk_home }}/ansible/roles/nginx/files/openwhisk-cert.pem
 whisk.ssl.key={{ openwhisk_home }}/ansible/roles/nginx/files/openwhisk-key.pem
 whisk.ssl.challenge=openwhisk
 whisk.api.host={{ whisk_api_host | default('https://' + (groups['edge'] | first)) }}
+whisk.system.id={{ whisk.system.id }}
+whisk.system.key={{ whisk.system.key }}
```
Comment (rabbah) [inline ansible/templates/whisk.properties.j2]: we need only the key, could look up the second, no?

File: tests/src/whisk/core/controller/test/AuthenticateTests.scala [hunk 2]
```diff
@@ -164,16 +164,16 @@ class AuthenticatedRouteTests
         }
     }
 
-    it should "report service unavailable when db lookup fails" in {
+    ignore should "report service unavailable when db lookup fails" in {
```
Comment (rabbah) [inline tests/src/whisk/core/controller/test/AuthenticateTests.scala]: ignored?

Comment (markusthoemmes) [inline tests/src/whisk/core/controller/test/AuthenticateTests.scala]: This is partly WIP. The condition that forced the internal error no longer holds water. I'll need to find a different way to trigger this.

File: ansible/templates/whisk.properties.j2 (hunk 1 above)
Comment (markusthoemmes) [inline ansible/templates/whisk.properties.j2]: Depends, will double-check if the lookup is only needed in tests, i'd say: yes.

File: tests/src/whisk/core/entity/test/DatastoreTests.scala [hunk 3]
```diff
@@ -77,16 +77,6 @@ class DatastoreTests extends FlatSpec
         putGetCheck(authstore, auth, WhiskAuth)
     }
 
-    it should "lookup key by uuid" in {
```
Comment (rabbah) [inline tests/src/whisk/core/entity/test/DatastoreTests.scala]: do we need a replacement test?

File: tests/src/whisk/core/controller/test/AuthenticateTests.scala (hunk 2 above)
Comment (rabbah) [inline tests/src/whisk/core/controller/test/AuthenticateTests.scala]: maybe I can cherry pick my mock data store :)

File: tests/src/whisk/core/entity/test/DatastoreTests.scala (hunk 3 above)
Comment (markusthoemmes) [inline tests/src/whisk/core/entity/test/DatastoreTests.scala]: I don't think so, functionality is gone.

We need testcoverage for the `Identity` object/class though.

**Comment by markusthoemmes** [comment]
PG2 889

**Comment by ddragosd** [comment]
I'm referencing the other PR #1715 which solves the problem with invoking actions without having to provide whisk credentials in case someone else stumbles about the same problem I did.

**Comment by rabbah** [comment]
Merged as https://github.com/openwhisk/openwhisk/compare/189968b19df1...ddee7fdcc820.

**Pull request closed without merging**